src/core/gltf2importer/gltf2parser_p.h

```diff
@@ -73,24 +77,32 @@ struct HierarchyNode {
 };
 
 using KeyParserFuncPair = QPair<QLatin1String, std::function<bool(const QJsonValue &)>>;
-
+class ThreadedGLTF2Parser;
 class KUESA_PRIVATE_EXPORT GLTF2Parser
+    : public QObject
 {
 public:
     GLTF2Parser(SceneEntity *sceneEntity = nullptr, bool assignNames = false);
     virtual ~GLTF2Parser();
 
     virtual QVector<KeyParserFuncPair> prepareParsers();
     Qt3DCore::QEntity *parse(const QString &filePath);
-    Qt3DCore::QEntity *parse(const QByteArray &data, const QString &basePath, const QString &filename = {});
```

Comment:
I would keep this one public as it'll be useful for supporting downloaded content